### PR TITLE
Added listener filter patching

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
For https://github.com/istio/istio/issues/41445

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@howardjohn

**Review by howardjohn — COMMENTED** [review]
Seems reasonable I suppose. We have a general problem of granularity, but ListenerFilter doe seem fairly common. @kyessenov @ramaraochavali any opinions?

**Review by howardjohn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@ericvn

File: networking/v1alpha3/envoy_filter.proto [hunk 1]
```diff
@@ -405,6 +405,30 @@ import "networking/v1alpha3/sidecar.proto";
 //                   local:
 //                     filename: "/etc/envoy_filter_http_wasm_example.wasm"
 // ```
+//
+// The following example insert a `envoy.filters.listener.proxy_protocol` listener filter before tls inspector.
```
Comment (ericvn) [inline networking/v1alpha3/envoy_filter.proto:409]: If we call out one filter specifically, should we call out the second?
```suggestion
// The following example inserts an `envoy.filters.listener.proxy_protocol` listener filter before the `envoy.filters.listener.tls_inspector`.
```

**Review by ericvn — APPROVED** [review]

**Pull request merged**